Azure/sonic-mgmt
# Fix the DUT-PTF port mismatch issue of topo_dualtor-56

### Description of PR


Summary:
Fixes # (issue)

Fix the DUT-PTF port mis-match issue of topo_dualtor-56

### Type of change




### Approach
#### What is the motivation for this PR?
The dualtor-56 topology has discontinuous sequence of host interfaces. For example,
interfaces 12,13,14,15 of DUT are used for upstream T1 connection. They are not vlan
interfaces. The vm_topology module for binding topology interfaces assumes that
the host interfaces are continuous sequence. This caused DUT-PTF port mismatch
issue.

#### How did you do it?
This change is to fix the issue by specifying host interfaces using format like
dut_index.dut_port_index@ptf_port_index. When ptf_port_index is explicitly specified,
vm_topology creates bridge and PTF interface using specified index instead of the
sequential 'i'. Same logic is applied to the function for removing host interfaces.
The function for validating topology file in vm_topology is also updated to support
the new host interface definition format. Related topo_facts module and testbed.py
library are updated as well.

#### How did you verify/test it?
* Add dualtor-56 topology
* Check 'minigraph_ptf_indices' in extended minigraph facts
* Check the ovs bridges and interfaces on server
* Run testbed-cli.sh deploy-mg
* Announce routes
* Remove dualtor-56 topology and add again

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation
